### PR TITLE
Add recipe for simulating synthetic lightcurves and fitting the PSD of an observed curve.

### PR DESCRIPTION
**Pull request by cgalelli**
This PR adds a recipe which implements some functions to generate synthetic lightcurves, nominally:
- the TK95 algorithm that will be added in the main gammapy code with 1.3
- the more complex EMM algorithm

It also adds the function to create envelopes of the periodogram for a lightcurve, and a fitting function for the PSD of an observed curve. These function can also be found in the small package [gammapy_SyLC](https://github.com/cgalelli/gammapy_SyLC) that will probably be adapted to be used here in the future, once problems with installation are resolved.

### CODE REVIEW

**Comment by bkhelifi** [comment]
Thanks a lot @cgalelli ...
@registerrier, should we precise in some headers the author, the date and the used Gammapy version?

**Review by registerrier — APPROVED** [review]
Thanks @cgalelli . This looks good to me. It would be simpler if some of the code could be imported for SyLC indeed. We can keep that for later though.

**Comment by registerrier** [comment]
> Thanks a lot @cgalelli ... @registerrier, should we precise in some headers the author, the date and the used Gammapy version?

The gammapy version is defined in the env.yml. Maybe a `print(gammapy.version)` would be sufficient.

Regarding names, we have not done it yet, but we could/should. What would be the proposal for that? How does additional changes (maintenance, upgrades) is taken into account?

**Comment by cgalelli** [comment]
I finally could solve the problem with the import of `gammapy_SyLC` so the new version has everything implemented as a package function. Changed the text accordingly. Also added the author line at the end.

**Review by registerrier — COMMENTED** [review]
Thanks @cgalelli . There seems to be a problem with the notebook. It is not recognized as such in the CI. Could you check if there is a problem? Is the very long name a good idea?

**Comment by cgalelli** [comment]
Thanks @registerrier , indeed i had made a bit of a mess in the restructuring of the notebook. Now it should be fine. I had also used an old local name for the notebook that was quite long and cumbersome -  returned to the name I had in the first commit.

The CI I think will still fail due to the import of `gammapy_SyLC`

File: recipes/fit-psd-lightcurve/env.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+channels:
+  - conda-forge
+
+dependencies:
+  - gammapy=1.2
+  - python=3.11
+  - scipy
+  - jupyter
+  - matplotlib
```
Comment (registerrier) [inline recipes/fit-psd-lightcurve/env.yml:13]:
```suggestion
  - matplotlib
  - pip:
    - git+https://github.com/cgalelli/gammapy_SyLC.git
```
Would this work?

**Comment by cgalelli** [comment]
Apparently it does not: i am trying to find an alternative way of installing the package in the environment file.

**Pull request merged**